src/core/ext/filters/message_size/message_size_filter.cc

```diff
@@ -117,17 +117,38 @@ void MessageSizeParser::Register() {
 
 size_t MessageSizeParser::ParserIndex() { return g_message_size_parser_index; }
 
+namespace {
+const grpc_arg* ChannelArgsFindLast(const grpc_channel_args* args,
+                                    const char* name) {
+  grpc_arg* arg = nullptr;
+  if (args != nullptr) {
+    for (size_t i = 0; i < args->num_args; ++i) {
+      if (strcmp(args->args[i].key, name) == 0) {
+        arg = &args->args[i];
+      }
+    }
+  }
+  return arg;
+}
+}  // namespace
+
 int GetMaxRecvSizeFromChannelArgs(const grpc_channel_args* args) {
   if (grpc_channel_args_want_minimal_stack(args)) return -1;
-  return grpc_channel_args_find_integer(
-      args, GRPC_ARG_MAX_RECEIVE_MESSAGE_LENGTH,
+  // We are not using grpc_channel_args_find_integer here because of ordering
```

Comment:
Rather than preserving the old broken behavior, I would prefer to fix the underlying problem.  We should never have more than one entry in channel args with the same key in the first place.  Let's try to fix things such that that does not happen.

Reply:
Updated. Made the change as discussed in b/159754453. Both the internal and external tests pass with this.

Reply:
Also running TGP to be safe